src/bin/dijkstra.rs

```diff
@@ -0,0 +1,203 @@
+// Dijkstra path planning
+// author: Salah Eddine Ghamri (s.ghamri)
+
+use gnuplot::*;
+use std::thread::sleep;
+use std::time::Duration;
+
+use itertools::iproduct;
+use std::collections::BTreeMap;
+
+use ordered_float::NotNan;
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+use na::DMatrix;
+extern crate nalgebra as na;
+use rust_robotics::grid_map;
+
+fn neighbors(nrows: usize, ncols: usize) -> impl Fn(usize, usize) -> Vec<(usize, usize)> {
```

Comment:
Consider adding a comment above the neighbors function like "The neighbors function returns adjacent cell coordinates for a given cell in a grid, excluding out-of-bound and the cell itself." for clarity.